packages/python/plotly/plotly/tests/test_core/test_errors/test_dict_path_errors.py

```diff
@@ -0,0 +1,154 @@
+
+
+# (Unfortunately I couldn't get the match argument of pytest.raises to work
+# for multiline regexes so we do the test with assert
+
+
+def test_raises_on_bad_index(some_fig):
+    # Check indexing errors can be detected when path used as key to go.Figure
+    try:
+        x0 = some_fig["layout.shapes[2].x0"]
+    except IndexError as e:
+        assert (
+            e.args[0].find(
+                """Bad property path:
+layout.shapes[2].x0
+              ^"""
+            )
+            >= 0
+        )
+
+
+def test_raises_on_bad_dot_property(some_fig):
+
+    # Check . property lookup errors can be detected when path used as key to
+    # go.Figure
+    try:
+        x2000 = some_fig["layout.shapes[1].x2000"]
+    except IndexError as e:
+        assert (
+            e.args[0].find(
+                """Bad property path:
+layout.shapes[1].x2000
+                 ^"""
+            )
+            >= 0
+        )
+
+    # Check . property lookup errors but not on the last part of the path
+    try:
+        x2000 = some_fig["layout.shapa[1].x2000"]
+    except IndexError as e:
+        assert (
+            e.args[0].find(
+                """Bad property path:
+layout.shapa[1].x2000
+       ^"""
+            )
+            >= 0
+        )
+
+
+def test_raises_on_bad_indexed_underscore_property(some_fig):
+
+    # finds bad part when using the path as a key to figure and throws the error
+    # for the last good property it found in the path
+    try:
+        some_fig["data[0].line_colr"] = "blue"
+    except ValueError as e:
+        assert (
+            (
+                e.args[0].find(
+                    """Bad property path:
+data[0].line_colr
+             ^"""
+                )
+                >= 0
+            )
+            and (
+                e.args[0].find(
+                    """Invalid property specified for object of type plotly.graph_objs.scatter.Line: 'colr'"""
+                )
+                >= 0
+            )
+        )
+
+    # finds bad part when using the path as a keyword argument to a subclass of
+    # BasePlotlyType and throws the error for the last good property found in
+    # the path
+    try:
+        some_fig.add_trace(go.Scatter(x=[1, 2], y=[3, 4], line_colr="blue"))
+    except ValueError as e:
+        assert (
+            (
+                e.args[0].find(
+                    """Bad property path:
+line_colr
+     ^"""
+                )
+                >= 0
+            )
+            and (
+                e.args[0].find(
+                    """Invalid property specified for object of type plotly.graph_objs.layout.shape.Line: 'colr'"""
+                )
+                >= 0
+            )
+        )
+
+    # finds bad part when using the path as a keyword argument to a subclass of
+    # BaseFigure and throws the error for the last good property found in
+    # the path
+    try:
+        fig2 = go.Figure(layout_title_txt="two")
+    except ValueError as e:
+        assert (
```

Comment:
for these assertions, it would be a lot more readable and future-proof to assert that the error message from `go.Figure(layout_title_txt="two")` is simply the same as the error message from `go.Figure(layout_title = dict(txt="two"))`, rather than doing assertions on the string itself, that way if and when we change the string we have fewer places to check :)

Reply:
This is another case for `with pytest.raisese`: https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions

Reply:
Hey Jon! I should close this PR, it's been superseded by this one: https://github.com/plotly/plotly.py/pull/2843

Reply:
Thanks!